src/foam/nanos/notification/email/EmailTemplateEngine.js

```diff
@@ -369,11 +364,12 @@ foam.CLASS({
       ps.setString(body);
       ParserContext parserX = new ParserContextImpl();
       parserX.set("sb", sbJoin);
+      parserX.set("emailTemplateDAO", x.get("emailTemplateDAO"));
       parserX.set("isNextTemplateExtending", false);
       getIncludeGrammar().parse(ps, parserX, "");
 
       if ( ! (Boolean) parserX.get("isNextTemplateExtending") ) return sbJoin;
-      return joinTemplates(sbJoin.toString());
+      return joinTemplates(sbJoin.toString(), x);
```

Comment:
Could you pass in the StringBuilder rather than converting to a string at each level?